src/Features/Core/Portable/UseCompoundAssignment/AbstractUseCompoundAssignmentCodeFixProvider.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.UseCompoundAssignment
+{
+    internal abstract class AbstractUseCompoundAssignmentCodeFixProvider<
+        TSyntaxKind, TAssignmentSyntax, TExpressionSyntax> 
+        : SyntaxEditorBasedCodeFixProvider
+        where TSyntaxKind : struct
+        where TAssignmentSyntax : SyntaxNode
+        where TExpressionSyntax : SyntaxNode
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+            ImmutableArray.Create(IDEDiagnosticIds.UseCompoundAssignmentDiagnosticId);
+
+        protected readonly ImmutableDictionary<TSyntaxKind, TSyntaxKind> BinaryToAssignmentMap;
+        protected readonly ImmutableDictionary<TSyntaxKind, TSyntaxKind> AssignmentToTokenMap;
+
+        protected AbstractUseCompoundAssignmentCodeFixProvider(
+            ImmutableDictionary<TSyntaxKind, TSyntaxKind> binaryToAssignmentMap,
+            ImmutableDictionary<TSyntaxKind, TSyntaxKind> assignmentToTokenMap)
+        {
+            BinaryToAssignmentMap = binaryToAssignmentMap;
+            AssignmentToTokenMap = assignmentToTokenMap;
+        }
+
+        protected abstract TSyntaxKind GetSyntaxKind(int rawKind);
```

Comment:
>GetSyntaxKind [](start = 39, length = 13)

Should `GetSyntaxKind` and/or `Token` be moved to `SyntaxFacts`?

Reply:
Let me see!